vm/src/obj/objiter.rs

```diff
@@ -61,13 +62,16 @@ fn iter_contains(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     );
     loop {
         match vm.call_method(&iter, "__next__", vec![]) {
-            Ok(element) => {
-                if &element == needle {
-                    return Ok(vm.new_bool(true));
-                } else {
-                    continue;
+            Ok(element) => match vm.call_method(needle, "__eq__", vec![element.clone()]) {
```

Comment:
Maybe you can reduce the amount of boilerplate here with the `?` operator. Please refer to here: https://stackoverflow.com/questions/42917566/what-is-this-question-mark-operator-about